piptools/scripts/compile.py

```diff
@@ -291,6 +305,27 @@ def _determine_linesep(
     help="Specify a package to consider unsafe; may be used more than once. "
     f"Replaces default unsafe packages: {', '.join(sorted(UNSAFE_PACKAGES))}",
 )
+@click.option(
+    "--build-distribution",
```

Comment:
Oh, and this one too — it needs to specify that it's about the build deps.

Reply:
Marking this as resolved since it is now hard to track in the code and I have made an attempt at addressing this.